packages/nodejs-scripts/src/currentSemSetup.ts

```diff
@@ -0,0 +1,64 @@
+import { CoverageTimeline, CURRENT_TERM, Holiday, semesterPeriods } from '@stex-react/utils';
+import { writeFile } from 'fs/promises';
+
+function isHoliday(date: Date, holidays: Holiday[]) {
+  const dateString = date.toISOString().split('T')[0];
+  return holidays.some((h) => h.date === dateString);
+}
+
+function currentSemSetup() {
+  const sem = semesterPeriods[CURRENT_TERM];
+  if (!sem) throw new Error(`Semester ${CURRENT_TERM} not found in semesterPeriods.`);
+  const holidays = sem.holidays;
+  const startDate = new Date(sem.lectureStart);
+  const endDate = new Date(sem.lectureEnd);
+  const courses = sem.courses;
+
+  const lectureEntriesByCourse: CoverageTimeline = {};
+  courses.forEach((c) => (lectureEntriesByCourse[c.courseId] = []));
+
+  const currentDate = new Date(startDate);
+  const dates = [];
+  while (currentDate <= endDate) {
+    dates.push(new Date(currentDate));
+    if (!isHoliday(currentDate, holidays)) {
```

Comment:
try to use early return pattern (or early continue in this case) to reduce nesting for both the `if` statements in this while-for loop